### PR TITLE
feat(search) Update the local search results cache when the user un/a…

### PR DESCRIPTION
**Pull request by CMasterson**
…rchives a search item

## Summary
Mutates the search results to remove the moved item

## References 
https://getpocket.atlassian.net/browse/IN-1315

## Implementation Details
When moving an item to or from the archive, also remove it from the current search results.

## Test Steps
Perform a search in your saves/archives
Move an item to the opposite store (Saves/Archives)
The item is removed from the search results

## PR Checklist:
- [ ] Added Unit / UI tests
- [ ] Self Review (review, clean up, documentation, run tests)
- [ ] Basic Self QA

### CODE REVIEW

File: PocketKit/Sources/PocketKit/MyList/SearchItemsList/SearchViewModel.swift [hunk 1]
```diff
@@ -445,12 +445,24 @@ extension SearchViewModel {
...
+        updateLocalSearchResults(ByRemoving: savedItem)
+    }
+
+    private func updateLocalSearchResults(ByRemoving item: SavedItem) {
+        if case let .searchResults(results) = searchState {
+            let updatedResults = results.filter {
+                $0.id != item.id
+            }
+
+            searchState = .searchResults(updatedResults)
```
Comment (CMasterson) [inline PocketKit/Sources/PocketKit/MyList/SearchItemsList/SearchViewModel.swift:464]: Needs a check for empty `updatedResults` array.

**Review by Gio2018 — DISMISSED** [review]
Looks good to me @CMasterson !

File: PocketKit/Sources/PocketKit/MyList/SearchItemsList/SearchViewModel.swift [hunk 2]
```diff
@@ -445,12 +445,24 @@ extension SearchViewModel {
...
         tracker.track(event: Events.Search.archiveItem(itemUrl: savedItem.url, positionInList: index, scope: selectedScope))
         source.archive(item: savedItem)
+        updateLocalSearchResults(ByRemoving: savedItem)
     }
 
     /// Triggers action to move an item from archive to saves in a list
     func moveToSaves(_ savedItem: SavedItem, index: Int) {
         tracker.track(event: Events.Search.unarchiveItem(itemUrl: savedItem.url, positionInList: index, scope: selectedScope))
         source.unarchive(item: savedItem)
+        updateLocalSearchResults(ByRemoving: savedItem)
```
Comment (cyndichin) [inline PocketKit/Sources/PocketKit/MyList/SearchItemsList/SearchViewModel.swift:455]: do we need to guard if the scope is in all items here?

**Review by timc-mozilla — APPROVED** [review]

**Review by cyndichin — APPROVED** [review]
looks good to me!

**Pull request merged**